EndpointHandler.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+require "settings.php";
+
+// Decodes the data array and returns the value of "claim"
+function getClaim($claim, $data) {
+	
+	$data_array = json_decode($data, true);
+	return $data_array[$claim];
+}
+
+// A class to handle both fetching and sending data to the various endpoints
+class EndpointHandler {
+	
+	private $metadata = "";
+	
+	public function __construct($policy_name) {
+		$this->getMetadata($policy_name);
+	}
+	
+	// Fetches the data at an endpoint using a HTTP GET request
+	public function getEndpointData($uri) {
+
+		$ch = curl_init($uri);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); 
+		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+		$resp = curl_exec($ch);
+		curl_close($ch);
+		return $resp;
+	}
+	
+	// Using a HTTP POST request, sends data to the endpoint and receives the response
+	public function postEndpointData($uri, $fields) {
+		
+		//url-ify the data for the POST
+		$fields_string = "";
+		foreach ($fields as $key=>$value) { $fields_string .= $key.'='.$value.'&'; }
```

Comment:
>$value [](start = 65, length = 6)

This is extremely dangerous because you're not ensuring `$value` is encoded already. Always do whatever you can to avoid building your own query strings. In PHP, you can use [http_build_query](http://php.net/manual/en/function.http-build-query.php). But this entire method should be replaced anyway with use of `wp_remote_post`, which obviates the need to build your own URL-encoded body anyway.